test/sets.jl

```diff
@@ -70,7 +70,7 @@ end
     @test ===(eltype(s3), Float32)
 end
 @testset "show" begin
-    @test sprint(show, Set()) == "Set{Any}()"
+    @test sprint(show, Set()) == "Set(Any[])"
```

Comment:
I'm not sure that change is an improvement. Maybe that allows simplifying the code, but `Set{Any}` is really the information that should be printed AFAICT.

Reply:
> `Set{Any}` is really the information that should be printed AFAICT.

Why do you think so? I find it more regular (and preferable) to always print `Set(array-form)` than resorting to `Set{T}()` only when it's empty, and as you said it simplifies the code.

Reply:
Ah, OK. I thought we printed e.g. `Set{Real}([1, 2]}`, but since that's not the case I guess it's fine.

Reply:
I think the printing may predate being able to write it that way, maybe we should change it to that instead?

Reply:
Indeed. Though probably better left for a separate PR.

Reply:
I got used to the printing `Set(array)` ! and on a not so serious note, it's more compact, we save 2 characters! But yes I would prefer to leave that to anther PR.